File: 7-kyu/upstream-downstream/python/solution.py
def time(distance, boat_speed, stream):
    direction, stream_speed = stream.split(' ')
    stream_speed = int(stream_speed)
    boat_speed += -stream_speed if direction[0] == 'U' else stream_speed
    t = round(distance / boat_speed, 2)
    return t

File: 7-kyu/upstream-downstream/typescript/solution.ts
export function time(distance: number, boatSpeed: number, stream: string): number{
  const [streamDir, streamSpeedRaw] = stream.split(" ");
  const streamSpeed = Number(streamSpeedRaw);
  const adjBoatSpeed = (boatSpeed += (streamDir[0] === "U" ? -streamSpeed : streamSpeed));
  return Number((distance / adjBoatSpeed).toFixed(2));
}